Add files via upload (#954)
plone/training: docs/content-editing/creating-managing-and-reordering-blocks.md

---
myst:
  html_meta:
    "description": "Creating, managing, and reordering blocks"
    "property=og:description": "Creating, managing, and reordering blocks"
    "property=og:title": "Creating, managing, and reordering blocks"
    "keywords": "Plone, Format, Block, Reorder"
---
# This guide offers a step-by-step approach to effectively create, manage, and reorder blocks within a Volto page.

1.  In this example, you'll edit a page.

    By default, the page includes a title block and the first text block.

>     To add another block, either press {kbd}`enter` or click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">.
> 
>     ````{card}
>     ```{image} _static/creating-managing-and-reordering-blocks-01.jpeg
>     :alt: Click on the "+" to add another block
>     :target: _static/creating-managing-and-reordering-blocks-01.jpeg

Reviewer: None of the images show up in https://plone-training--954.org.readthedocs.build/content-editing/creating-managing-and-reordering-blocks.html. I'll edit these, and leave the rest to you to fix.

`````suggestion
    To add another block, either press {kbd}`enter` or click on the {guilabel}`Add block` button <img alt="Add block icon" src="/content-editing/_static/circle-plus.svg" class="inline">.

    ````{card}
    ```{image} /content-editing/_static/creating-managing-and-reordering-blocks-01.jpeg
    :alt: Click on the "+" to add another block
    :target: /content-editing/_static/creating-managing-and-reordering-blocks-01.jpeg
`````

    ```
    +++
    _To add another block, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">._
    ````

1.  Click on the type of block you want to add. For this example you'll add a few images and some text.
        
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-02.jpeg

>     :alt: Add type of block you want

Reviewer: ```suggestion
    :alt: Add the type of block you want
```

    :target: _static/creating-managing-and-reordering-blocks-02.jpeg
    ```
    +++
    _Add the type of block you want_
    ````

> 1.  Manage a block in the block itself. The Image block allows you to browse for an image or upload one.

Reviewer: "image block" is not a proper noun, and its context is not a GUI label, so it should be lower case.

```suggestion
1.  Manage a block in the block itself.
    The image block allows you to browse for an image or upload one.
```

    
    Most blocks also provide options in the sidebar under the {guilabel}`block` tab.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-03.jpeg
    :alt: Manage block options in block
    :target: _static/creating-managing-and-reordering-blocks-03.jpeg
    ```
    +++
    _Manage block options in block_
    ````

> 1.  For the Image block, use the sidebar to browse for an image in the site.

Reviewer: ```suggestion
1.  For the image block, use the sidebar to browse for an image in the site.
```

    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-04.jpeg
    :alt: Manage block options in sidebar
    :target: _static/creating-managing-and-reordering-blocks-04.jpeg
    ```
    +++
    _Manage block options in sidebar_
    ````

1.  To add another block, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-05.jpeg
    :alt: To add another block, press the "+" button
    :target: _static/creating-managing-and-reordering-blocks-05.jpeg
    ```
    +++
    _To add another block, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">._
    ````

> 1.  For this example you'll add a {guilabel}`Grid` block to add a few images.

Reviewer: ```suggestion
1.  For this example you'll add a grid block to add a few images.
```

    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-06.jpeg
    :alt: Add a grid block
    :target: _static/creating-managing-and-reordering-blocks-06.jpeg
    ```
    +++

>     _Add a {guilabel}`Grid` block_

Reviewer: ```suggestion
    _Add a grid block_
```

    ```
    +++
    _Add an image in each column_
    ````

1.  After finishing the columns, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-10.jpeg
    :alt: To add another block, press the "+" button
    :target: _static/creating-managing-and-reordering-blocks-10.jpeg
    ```
    +++
    _To add another block, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">._
    ````

1.  Add some text.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-11.jpeg
    :alt: Add some text
    :target: _static/creating-managing-and-reordering-blocks-11.jpeg
    ```
    +++
    _Add some text_
    ````

1.  Click inside the text block, then paste or type.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-12.jpeg
    :alt: Click inside the text block and paste or type
    :target: _static/creating-managing-and-reordering-blocks-12.jpeg
    ```
    +++
    _Click inside the text block and paste or type_
    ````

> 1.  After adding text, you will reorder the blocks.

Reviewer: ```suggestion
1.  After adding text, you'll reorder the blocks.
```

    ```
    +++
    _Reorder the blocks_
    ````

1.  To reorder, click the handle to the left of a block <img alt="Add block icon" src="_static/block-handle.svg" class="inline">, then drag the block up or down.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-14.jpeg
    :alt: Reorder blocks with toggle on block
    :target: _static/creating-managing-and-reordering-blocks-14.jpeg
    ```
    +++
    _Reorder blocks with toggle on block_
    ````

1.  Once it is in the position you want, release the mouse button.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-15.jpeg
    :alt: Drag and drop blocks
    :target: _static/creating-managing-and-reordering-blocks-15.jpeg
    ```
    +++
    _Drag and drop blocks_
    ````

1.  You can also change the order under the {guilabel}`Order` tab.
    
    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-16.jpeg
    :alt: Change order of blocks in "Order" tab
    :target: _static/creating-managing-and-reordering-blocks-16.jpeg
    ```
    +++
    _Change order of blocks in {guilabel}`Order` tab_
    ````

> 1.  In the {guilabel}`Order` tab, you can drag the items in the list up and down to reorder the blocks on the page.

Reviewer: ```suggestion
1.  In the {guilabel}`Order` tab, you can drag and drop the items in the list up and down to reorder the blocks on the page.
```

---
myst:
  html_meta:
    "description": "Creating, managing, and reordering blocks"
    "property=og:description": "Creating, managing, and reordering blocks"
    "property=og:title": "Creating, managing, and reordering blocks"
    "keywords": "Plone, Format, Block, Reorder"
---

# This guide offers a step-by-step approach to effectively create, manage, and reorder blocks within a Volto page.

> 1. In this example, we're editing a page.
> 
> By default it will already include the title and first text block.
> 
> To add more text, simply press enter. Or click on the "+" to add another block.
> 
> 
> 
> ````{card}
> ```{image} _static/creating,-managing,-and-reordering-blocks-01.jpeg
> :alt: Click on the "+" to add another block
> :target: _static/creating,-managing,-and-reordering-blocks-01.jpeg
> ```
> +++
> _Click on the "+" to add another block_
> ````

Reviewer: Content within enumerated lists must be indented to maintain its context, else content appears to be a list item followed by a paragraph and a screenshot.

Images and links (targets) must have valid references.

Enumerations should use `1.`, in case the enumeration of steps changes. MyST knows how to count, so you don't have to.

Also it's a good idea to indent the first enumerated item to the fourth space to make it easier for code editors to recognize consistent indentation. Some editors will incorrectly indent nested blocks. See https://stackoverflow.com/a/48313531/2214933 and https://stackoverflow.com/a/45683081/2214933.

Please use one sentence per line. See [Plone documentation styleguide](https://6.docs.plone.org/contributing/documentation/authors.html#plone-documentation-styleguide) for details.

I also enhanced the markup for keystrokes and icons. See [Inline images](https://6.docs.plone.org/contributing/documentation/myst-reference.html#inline-images).

Avoid words like "simply" or "easy", because if the reader does not find something to be simple or easy, they may feel badly. Web applications are never simple.

Avoid using "we", preferring to use "you" when addressing the reader. See [Person](https://learn.microsoft.com/en-us/style-guide/grammar/person). In an in-person tutorial, "we" may apply, but most of the time, the reader will do the work alone. I'll leave it to you to decide which voice is more appropriate for this tutorial.

For verb tense, see [Verbs](https://learn.microsoft.com/en-us/style-guide/grammar/verbs).

I've done the first item, and I'll leave the rest for you to clean up.

`````suggestion
1.  In this example, you'll edit a page.

    By default, it will already include a title block and the first text block.

    To add another block, either press {kbd}`enter` or click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">.

    ````{card}
    ```{image} _static/creating-managing-and-reordering-blocks-01.jpeg
    :alt: Click on the "+" to add another block
    :target: _static/creating-managing-and-reordering-blocks-01.jpeg
    ```
    +++
    _To add another block, click on the {guilabel}`Add block` button <img alt="Add block icon" src="_static/circle-plus.svg" class="inline">._
    ````
`````
